packages/headless/src/features/generated-answer/generated-answer-selectors.ts

```diff
@@ -1,12 +1,21 @@
 import {createSelector} from '@reduxjs/toolkit';
+import {
+  selectAnswer,
+  StateNeededByAnswerAPI,
+} from '../../api/knowledge/stream-answer-api';
 import {GeneratedAnswerCitation} from '../../controllers/generated-answer/headless-generated-answer';
 import {SearchAppState} from '../../state/search-app-state';
 import {GeneratedAnswerSection} from '../../state/state-sections';
 import {selectQuery} from '../query/query-selectors';
 
 export const generativeQuestionAnsweringIdSelector = (
-  state: Partial<SearchAppState>
-) => state.search?.response?.extendedResults?.generativeQuestionAnsweringId;
+  state: Partial<SearchAppState> | StateNeededByAnswerAPI
+) => {
```

Comment:
`Partial<SearchAppState>` is a "very large/lax" type which means there's lots of nullish things to check for. So that might be the problem in hindsight. However, I would not extend it even more with `StateNeededByAnswerAPI` because those overlap anyway.

Here is a suggestion

```suggestion
export const generativeQuestionAnsweringIdSelector = (
  state: Partial<SearchAppState>
) => {
  if (isSearchSection(state)) {
    return state.search.response.extendedResults.generativeQuestionAnsweringId;
  }
  if (isGeneratedAnswerSection(state)) {
    return selectAnswer(state).data?.answerId;
  }

  return undefined;
};

const isSearchSection = (
  state: Partial<SearchAppState> | StateNeededByAnswerAPI
): state is SearchSection => 'search' in state;

const isGeneratedAnswerSection = (
  state: Partial<SearchAppState>
): state is StateNeededByAnswerAPI =>
  'answer' in state &&
  'generatedAnswer' in state &&
  !isNullOrUndefined(state.generatedAnswer?.answerConfigurationId);
```

Reply:
Thanks, I'll give it a shot !

Reply:
Never found any good use case for the `is` operator in TS.  This type of narrowing is perfect. Thanks